include/battle.h

```diff
@@ -655,6 +655,7 @@ struct BattleStruct
     u16 overwrittenAbilities[MAX_BATTLERS_COUNT];    // abilities overwritten during battle (keep separate from battle history in case of switching)
     bool8 allowedToChangeFormInWeather[PARTY_SIZE][2]; // For each party member and side, used by Ice Face.
     u8 battleBondTransformed[NUM_BATTLE_SIDES]; // Bitfield for each party.
+    bool8 hitSwitchTargetFailed;
```

Comment:
Since it's just TRUE/FALSE, maybe it could be changed to one bit to save EWRAM space?

Reply:
I think `gBattleStruct` is always in EWRAM isn't it? (via being `AllocZeroed` on the heap)